podkrepi-bg/frontend
# New Donation Flow -  Choosing a payment method step (Card with Stripe)

Closes #1284 

## Motivation and context
Started implementation on the new Donation Flow with initilizing the Stripe Elements Payment Details form.

* Created a new folder and a page `/donation-v2` to temporarily hold the new donation flow. And a `donation-flow` components folder for all the new implementations
* Created a `DonationFlowContext` to handle the Stripe Payment Intent and information needed for the submission of the donation that is not part of the `Formik` context
* Styled and created the RadioGroups for mobile and desktop for choosing a `PaymentMethod`
* Reused components such as the picking of the price and tax calculation checkboxes from the previous implementation

## Screenshots:

Mobile|Desktop
---|---
![stripe-setup-flow-mobile](https://user-images.githubusercontent.com/61479393/212423260-a19d1eca-8c20-474d-b598-6cf23b6da2af.gif)|![stripe-setup-flow-desktop](https://user-images.githubusercontent.com/61479393/212423258-59a93ae9-4020-4428-b169-0c42a67fff92.gif)



## Testing

### Steps to test
1.  Go to `/campaigns/donation-v2/{slug}`
2. Pick a price from the predefined prices 
3. Observe and test the stripe initilized card details form
### Affected urls
`/campaigns/donation-v2/{slug}`





### New or updated dependencies:



Dependency name|Previous version|Updated version|Details
---|---|---|---
`next`|`v13.0.5`|`v13.1.0`|`modularizeImports` in the config is not experimental anymore
